src/game/common/system/asciistring.cpp

```diff
@@ -100,29 +104,33 @@ void Utf8String::Free_Bytes()
 void Utf8String::Ensure_Unique_Buffer_Of_Size(
     int chars_needed, bool keep_data, const char *str_to_cpy, const char *str_to_cat)
 {
+    Validate();
+
     if (m_data != nullptr && m_data->ref_count == 1 && m_data->num_chars_allocated >= chars_needed) {
         if (str_to_cpy != nullptr) {
             strcpy(Peek(), str_to_cpy);
         }
 
         if (str_to_cat != nullptr) {
-            strcpy(Peek() + strlen(Peek()), str_to_cat);
+            strcat(Peek(), str_to_cat);
         }
 
     } else {
-        // this block would have been a macro like DEBUG_CRASH(numCharsNeeded + 8 > MAX_LEN, THROW_02); (see cl_debug.h)
-        // if ( numCharsNeeded + 8 > MAX_LEN ) {
-        // *&preserveData = 0xDEAD0002;
-        // throw(&preserveData, &_TI1_AW4ErrorCode__);
-        //}
+#ifdef GAME_DLL
+        static_assert(sizeof(AsciiStringData) == 4);
+#endif
+        const int required_size = chars_needed + sizeof(AsciiStringData);
+
+        if (required_size > MAX_LEN) {
```

Comment:
Can't this just be a captains log rel assert

Reply:
Ok. Changed to relassert.